### ISSUE TITLE
better support for when .flowconfig isn't at project root

### ISSUE THREAD

**Issue by chrisdrackett** [AtomLinter/linter-flow#43]
Right now if you are only using flow in a sub-directory of your project the linter will fail. Ideally when opening a file the linter would find the .flowconfig file in the nearest parent directory, start a flow server for that directory if one isn't already open, then poll the flow server for errors in that directory.

**Comment by Adezandee** [comment]
Hi @chrisdrackett 
That's a concern that should be raised in `atom-linter`, because we are already looking for .flowconfig in parent folders with `atomLinter.findFile(filePath, '.flowconfig')`

**Comment by steelbrain** [comment]
@Adezandee `atom-linter` already searches for given file up to root

**Comment by Arcanemagus** [comment]
Specifically it searches for it [here](https://github.com/AtomLinter/linter-flow/blob/v4.0.1/lib/index.js#L86), and ensures the running server has that path [here](https://github.com/AtomLinter/linter-flow/blob/v4.0.1/lib/servers.js#L15).

Do you have an example directory structure where this isn't working properly?

**Comment by chrisdrackett** [comment]
maybe I'm having another issue. When I start the editor I get a blue notification that flow has started (and I can see it running) then I get a red one saying that flow could not be spawned.

**Comment by nmn** [comment]
My best guess is that atom couldn't find the flow executable. Try setting the path to the flow executable in the linter-flow settings.

Use this to find the path:

```sh
$ which flow
```
